### PR TITLE
Visibility of the chat message resolution

### PR DESCRIPTION
**Pull request by herveDarritchon**
Visibility of the chat message opposed roll resolution is based on the initial message visibility (defender, attacker). Now, you should not see the resolution if the roll mode is GM/Blind GM.

### CODE REVIEW

File: scripts/opposed-wfrp4e.js [hunk 1]
```diff
@@ -270,23 +278,35 @@ class OpposedWFRP
...
       hideData: true,
+      rollMode: rollMode,
       flags:{"opposedStartMessage": true},
       content: `<div><b>${speaker.alias}<b> ${game.i18n.localize("ROLL.OpposedStart")}<div>`
-    }).then(msg => this.startMessage = msg)
+    }
+
+    if (["gmroll", "blindroll"].includes(chatOptions.rollMode)) chatOptions["whisper"] = ChatMessage.getWhisperIDs("GM");
+    if (chatOptions.rollMode === "blindroll") chatOptions["blind"] = true;
+    else if (chatOptions.rollMode === "selfroll") chatOptions["whisper"] = [game.user];
```
Comment (moo-man) [inline scripts/opposed-wfrp4e.js]: Can you use `chatDataSetup` to avoid duplicating this code?

File: scripts/opposed-wfrp4e.js [hunk 2]
```diff
@@ -367,7 +387,9 @@ class OpposedWFRP
         await OpposedWFRP.evaluateOpposedTest(attacker, defender,
         {
           target: true,
-          startMessageId: actor.data.flags.oppose.startMessageId
+          startMessageId: actor.data.flags.oppose.startMessageId,
+          whisper: message.data.whisper,
+          blind: message.data.blind,
```
Comment (moo-man) [inline scripts/opposed-wfrp4e.js:380]: Similarly, can you just pass `rollMode` and use `chatDataSetup` instead?

File: scripts/opposed-wfrp4e.js (hunk 1 above)
Comment (herveDarritchon) [reply scripts/opposed-wfrp4e.js]: Ok.

File: scripts/opposed-wfrp4e.js (hunk 2 above)
Comment (herveDarritchon) [reply scripts/opposed-wfrp4e.js:380]: In that case, it is not the rollMode activated in the GM client. It is the whisper and blind value from the modified chat message. So for me, using rollMode and chatDataSetup will not have the same result.

**Comment by herveDarritchon** [comment]
@moo-man I have one conversation in stand by. What do you think ?

File: scripts/opposed-wfrp4e.js [hunk 3]
```diff
@@ -80,16 +81,17 @@ class OpposedWFRP
...
   /**
    * Defender responds to an opposed test - evaluate result
-   * 
+   *
    * @param {Object} testResult Test result values
    * @param {Object} message message for update, actor, token, etc. for the defender
+   * @param {String} rollMode the type of roll mode used for the card
    */
-  static defenderClicked(testResult, message)
+  static defenderClicked(testResult, message, rollMode)
```
Comment (moo-man) [inline scripts/opposed-wfrp4e.js]: rollMode doesn't appear to be used in this function

File: scripts/opposed-wfrp4e.js [hunk 4]
```diff
@@ -269,23 +276,27 @@ class OpposedWFRP
...
   }
 
-  // Update starting mesasge with result - manual opposed
+  // Update starting message with result - manual opposed
   static updateOpposedMessage(damageConfirmation, msgId)
   {
-    let opposeMessage = game.messages.get(msgId)
+    let opposeMessage = game.messages.get(msgId);
+    let rollMode=opposeMessage.data.rollMode;
+    console.log(`Update - Roll Mode selected : ${rollMode}`);
```
Comment (moo-man) [inline scripts/opposed-wfrp4e.js]: I don't think this console.log is needed outside debugging purposes

File: scripts/utility-wfrp4e.js [hunk 5]
```diff
@@ -612,6 +612,11 @@ class WFRP_Utility
     if (chatData.rollMode === "blindroll") chatData["blind"] = true;
     else if (chatData.rollMode === "selfroll") chatData["whisper"] = [game.user];
 
+    if ( forceWhisper ) { // Final force !
+      chatData["speaker"] = ChatMessage.getSpeaker();
```
Comment (moo-man) [inline scripts/utility-wfrp4e.js]: Why is `getSpeaker()` necessary here? I tend to avoid this function if I can.

File: scripts/market-wfrp4e.js [hunk 6]
```diff
@@ -437,31 +437,52 @@ class MarketWfrp4e {
         {
             let amount
             let nbActivePlayers = getNbOfActivePlayers();
-
+            let forceWhisper
+            
+            console.log("This is my option", option);
```
Comment (moo-man) [inline scripts/market-wfrp4e.js:442]: Unnecessary debug log

File: scripts/opposed-wfrp4e.js (hunk 3 above)
Comment (herveDarritchon) [reply scripts/opposed-wfrp4e.js]: yes, not use anymore.

File: scripts/opposed-wfrp4e.js (hunk 4 above)
Comment (herveDarritchon) [reply scripts/opposed-wfrp4e.js]: what is your strategy about logging ? only errors but sometimes I see some debugging stuff.

File: scripts/utility-wfrp4e.js (hunk 5 above)
Comment (herveDarritchon) [reply scripts/utility-wfrp4e.js]: I don't know, not my stuff :( @LeRatierBretonnien.
Moreover getSpeaker takes 1 parameter and this is not the case.
I can delete it if you think so.

**Comment by herveDarritchon** [comment]
since you have evolved in the repo, I have made a rebase of my branch to follow your commits.

File: scripts/opposed-wfrp4e.js (hunk 4 above)
Comment (moo-man) [reply scripts/opposed-wfrp4e.js]: Debugging stuff is probably meant to be deleted, I know Edmund has a bunch of odd logging that I haven't deleted yet. The only general log wfrp4e system does is when it plays audio, if I recall

File: scripts/utility-wfrp4e.js (hunk 5 above)
Comment (moo-man) [reply scripts/utility-wfrp4e.js]: Oh sorry, I thought you added the forceWhisper condition

**Pull request merged**